api/src/main/java/org/openmrs/module/reporting/report/service/ReportServiceImpl.java

```diff
@@ -479,6 +485,26 @@ public Report runReport(ReportRequest request) {
 				renderReportDataToBytes = !(renderer instanceof InteractiveReportRenderer);
 			}
 
+			for (Map.Entry<String, DataSet> e : reportData.getDataSets().entrySet()) {
```

Comment:
The dates from the resutSet are being returned as LocalDateTime, not Date. Not sure if this happens due to Java or jdbc driver version we're using or even if it may return some other type.
Here, I'm converting LocalDateTime to Date considering the timezone on the request so that field is treated as Date on the report.

Reply:
Could we add the above to the code?

Reply:
I don't think this is the ideal place to put this code.  This feels more like something that should be handled at either the time of rendering or the time of evaluation, than something to apply like this as post-processing to an entire report within the service.

If we go with the 2nd option I propose above, which is to have clientTimezone able to be defined as a context value in the evaluation context and overridden by a parameter value in the report request, then the next step would be to create a generic method to render out column values that are of type LocalDate by taking into account the clientTimezone set in the Evaluation Context contextValues.

Reply:
Regarding the place to do this conversion, I'm kind of divided on whether doing it at the time of evaluation or rendering.

I assume that, timezone conversion at the time of evaluation, for the SQL, would be done [here](https://github.com/openmrs/openmrs-module-reporting/blob/c0264d4cc441685edf1b3332bfd464d69c8f2e42/api/src/main/java/org/openmrs/module/reporting/dataset/definition/evaluator/SqlDataSetEvaluator.java#L92
) but I don't know about the other evaluators and scenarios. Conversion should be handled on all evaluators other than `SqlDataSetEvaluator`, no?

If done at the time of rendering would I assume it would be done on the [XlsReportRenderer](https://github.com/openmrs/openmrs-module-reporting/blob/c0264d4cc441685edf1b3332bfd464d69c8f2e42/api/src/main/java/org/openmrs/module/reporting/report/renderer/XlsReportRenderer.java#L74) for the Excel but, again, it would have to be done for all the renders, right?

I don't have a definitive opinion on this. @mseaton @ibacher could you provide more input on your thoughts?

Reply:
Great points, and I agree that this is tricky (see my comment above).  Probably this does not make the most sense.